backend/src/routes/Call.ts

```diff
@@ -1,28 +1,89 @@
 import express, { Router } from 'express';
 import * as z from 'zod';
+import _ from 'lodash';
 import { requireSelf } from './middlewares';
 import type {
   Call as CallService,
   TwilioCall as TwilioCallService,
+  Contact as ContactService,
 } from '../services';
 import Stopwatch from '../util/stopwatch';
 import {
   validateRequest,
   stringToNumberTransformer,
 } from './helpers/validation';
+import { Call, Contact } from '../models';
+import TwilioCall from '../models/TwilioCall';
 
 const GET_SCHEMA = z.object({
   params: z.object({
     userId: stringToNumberTransformer,
   }),
 });
 
+const GET_RECENT_SCHEMA = z.object({
+  params: z.object({
+    userId: stringToNumberTransformer,
+  }),
+});
+
+function mapObjectsToRecentCallResponse(
+  calls: Call[],
+  contacts: Contact[],
+  twilioCalls: TwilioCall[]
+) {
+  return calls.map((call) => {
+    const currentContact = contacts.find(
+      (contact) => contact.id === call.contactId
+    );
+    const currentTwilioCall = twilioCalls.find(
+      (twilioCall) => twilioCall.twilioSid === call.incomingTwilioCallSid
+    );
+    return {
+      id: call.id,
+      avatar: _.get(currentContact, 'avatar', null),
```

Comment:
Looks like the type is inferred properly. I will change it since we are using keys from a single level.

<img width="656" alt="Screenshot 2020-11-25 at 9 02 31 AM" src="https://user-images.githubusercontent.com/3167732/100168848-01660980-2efd-11eb-9609-b8552a40a31a.png">